src/method.c

```diff
@@ -297,10 +307,60 @@ static void jl_code_info_set_ir(jl_code_info_t *li, jl_expr_t *ir)
             else
                 jl_array_del_end(meta, na - ins);
         }
+        // check other flag expressions
+        else if (jl_is_expr(st) && ((jl_expr_t*)st)->head == inbounds_sym) {
+            is_flag_stmt = 1;
+            jl_array_t *args = ((jl_expr_t*)st)->args;
+            jl_value_t *arg1 = jl_array_ptr_ref(args, 0);
+            if (arg1 == (jl_value_t*)jl_true)       // push
+                inbounds_depth += 1;
+            else if (arg1 == (jl_value_t*)jl_false) // clear
+                inbounds_depth = 0;
+            else if (inbounds_depth > 0)            // pop
+                inbounds_depth -= 1;
+        }
```

Comment:
```suggestion
            bd[j] = jl_nothing;
        }
```
usually after processing metadata, we remove it

Reply:
I think it's still useful for debugging to keep them in lowered form ?